### ISSUE TITLE
Support for loading & saving of ascii-armored keys

### ISSUE THREAD

**Issue by ribose-jeffreylau** [riboseinc/ruby-rnp#16]
Hi @dewyatt , is there support for import / export of ascii-armored keys?  Thanks!

**Comment by ni4** [comment]
@ribose-jeffreylau If you build directly from maste then they should be loadable like ordinary keyring (at least for the single armored key).

**Comment by ribose-jeffreylau** [comment]
@ni4 Thanks!  So what's left to implement is the exporting of keys, is that correct?

**Comment by dewyatt** [comment]
@ribose-jeffreylau Correct, I haven't exposed that functionality within `ruby-rnp` yet.

**Comment by dewyatt** [comment]
@ribose-jeffreylau Ok I can prioritize this. I'll take a look and let you know.

**Comment by ribose-jeffreylau** [comment]
@dewyatt That's awesome, thanks!

**Comment by dewyatt** [comment]
I'm still working on some tests and finishing things up, so it'll probably be another day.

**Comment by dewyatt** [comment]
Well I had a run-in with the `rm -rf` monster from my command history, and my backup system wasn't in place, so I have to search for a few source files hiding on disk and resume from there. :(

**Comment by ronaldtse** [comment]
It’s something that happens to everyone, so it’s just damage control afterwards. Reminds us that one can still improve oneself :-)

**Comment by dewyatt** [comment]
Thanks @ronaldtse. :)
I managed to recover the key parts and got my backup system back in place again.

@ribose-jeffreylau 
I think the reason I didn't implement this initially was that I wasn't satisfied with the functionality on the rnp end, so we are going to be waiting on https://github.com/riboseinc/rnp/pull/732 for this.
I will continue on the ruby side until then.

**Comment by dewyatt** [comment]
This should be good, just waiting on reviews. It's in the [dewyatt-key-export](/riboseinc/ruby-rnp/tree/dewyatt-key-export) branch, but it requires the [dewyatt-ffi-key-export](/riboseinc/rnp/tree/dewyatt-ffi-key-export) branch of rnp too. Hopefully it will be merged soon.

**Comment by ribose-jeffreylau** [comment]
@dewyatt Hi, sorry to hear about the loss... but glad to hear about the backup system :)

 The changes look good!  Thanks!

**Comment by ribose-jeffreylau** [comment]
Cheers @dewyatt !
